proposals/NNNN-high-level-interface-foundation.md

```diff
@@ -0,0 +1,421 @@
+
+![view-controller-lifecycle](../assets/proposals/NNNN-high-level-interface-foundation/view-controller-lifecycle.png)
+
+![view-controller-overview](../assets/proposals/NNNN-high-level-interface-foundation/view-controller-overview.png)
+
+#### `Init`
+
+View controllers are instantiated when they are pushed to the view controller manager. In the case of the root view controller it would be instantiated once a connection is established to the head unit (app is registered). In this state there is no view loaded. The app developer should not load data or perform other memory/time intense code at this time.
+
+#### `loadView`
+
+In case the view controller appears on the screen but does not have a loaded view the method `loadView` is called which should be overriden by the app developer in the subclass.
+
+#### `viewDidLoad`
+
+After calling `loadView` the view controller is notified that the view did load. The app developer can override this method in order to load data sources and apply assets to the view.
+
+#### `viewWillAppear`
+
+Whenever a view controller's view should become visible on the head unit the view controller will be notified that the view will appear. Reasons to appear on the head unit are:
+1. The app launched and became (in)active (entered HMI_LIMITED or HMI_FULL). The root view controller will appear.
+2. The app has pushed a second view controller to the stack. The second view controller's view will appear after the first view controller's view will disapper.
+3. The top view controller was removed from the stack. The underlying view controller's view will appear after the top view controller's view will disappear.
+
+The app developer can override this method in order to manipulate view elements before they are presented on the head unit.
+
+#### `viewDidAppear`
+
+This method is called after `viewWillAppear` and after the view related RPCs (`Show`, `SetDisplayLayout` etc.) are all performed.
+
+#### `viewWillDisappear`
+
+Whenever a view is/will not be visible on the head unit anymore the view controller will be notified by `viewWillDisappear`. Reasons for that are:
+1. The app entered HMI_NONE or HMI_BACKGROUND. In this case the view already disappeared. Still in order to follow the flow this method should be called.
+2. The app has pushed a second view controller to the stack. The first view controller's view will disappear before the second view controller's view will appear.
+3. The top view controller was removed from the stack. The top view controller's view will disappear before the underlying view controller's view will appear.
+
+#### `viewDidDisappear`
+
+This method is called after `viewWillDisappear`. 
+
+### SDLViewController
+
+The base class provides empty overridable methods in order to load a view or being notified when a view did load or if will/did appear/disappear. 
```

Comment:
Please update to "...or to be notified when a view did load or if it will/did appear/disappear."